### ISSUE TITLE
Needless "if let" body in if let assignment

### ISSUE THREAD

**Issue by lesurp** [rust-lang/rust-clippy#10501]
### What it does

One can avoid using the body of an "if" block for destructuring e.g. an Option by simple pattern matching:
```rust
    let a = Some(1u8);
    let Some(b) = a else { panic!() };
```

However this is not always obvious, and is sometimes written as:
```rust
    let a = Some(1u8);
    let b = if let Some(a) = a { a } else { panic!() };
```
which is less readable.

### Lint Name

needless_if_let_body

### Category

style

### Advantage

- makes code easier to read 
- prevents potential mistakes (but if "a" is unused in my example, a unused-variable warning would be thrown anyway)

### Drawbacks

None AFAIK, this is purely style-related.
If anything is done inside the `if let` besides returning the contained value then this lint should stay silent.

### Example

```rust
    let a = Some(1u8);
    let b = if let Some(a) = a { a } else { panic!() };
```

Could be written as:

```rust
    let a = Some(1u8);
    let Some(b) = a else { panic!() };
```

**Comment by Alexendoo** [comment]
This exists as [`manual_let_else`](https://rust-lang.github.io/rust-clippy/master/index.html#manual_let_else), it's currently in `pedantic` (off by default) but will likely migrate to `style` once rustfmt support is there for let-else
